apps/server/src/modules/room/api/room.uc.ts

```diff
@@ -342,16 +348,45 @@ export class RoomUc {
 		});
 	}
 
-	private mapToMember(member: UserWithRoomRoles, user: UserDo): RoomMemberResponse {
-		const schoolRoleNames = user.roles.map((role) => role.name);
-		return new RoomMemberResponse({
-			userId: member.userId,
-			firstName: user.firstName,
-			lastName: user.lastName,
-			roomRoleName: member.roles[0].name,
-			schoolRoleNames,
-			schoolName: user.schoolName ?? '',
+	private buildRoomMembersResponse(
+		users: UserDo[],
+		roomMembershipAuthorizable: RoomMembershipAuthorizable
+	): RoomMemberResponse[] {
+		const membersResponse = users.map((user) => {
+			const member = roomMembershipAuthorizable.members.find((item) => item.userId === user.id);
+			if (!member) {
+				/* istanbul ignore next */
+				throw new Error('User not found in room members');
+			}
+			const schoolRoleNames = user.roles.map((role) => role.name);
+			return new RoomMemberResponse({
+				userId: member.userId,
+				firstName: user.firstName,
+				lastName: user.lastName,
+				roomRoleName: member.roles[0].name ?? '',
+				schoolRoleNames,
+				schoolName: user.schoolName ?? '',
+				schoolId: user.schoolId,
+			});
+		});
+		return membersResponse;
+	}
+
+	private handleAnonymization(
+		membersResponse: RoomMemberResponse[],
+		currentUserSchoolId: EntityId
+	): RoomMemberResponse[] {
+		const anonymizedMembersResponse = membersResponse.map((member) => {
+			const isRoomOwner = member.roomRoleName === RoleName.ROOMOWNER;
+			const isCurrentUserInSameSchool = member.schoolId === currentUserSchoolId;
+			const shouldBeAnonymized = !isRoomOwner && !isCurrentUserInSameSchool;
```

Comment:
```suggestion
			const isFromSameSchool = member.schoolId === currentUserSchoolId;
			const shouldBeAnonymized = !isRoomOwner && !isFromSameSchool;
```
a little easier to read

Reply:
very good suggestion!